### PR TITLE
Add setGroupResourceAccess

### PR DESCRIPTION
**Pull request by NSeydoux**
This adds setGroupResourceAccess to the WAC API. This function sets access for a
given group and a given resource if the resource is controlled using
WAC. This only applies to Resource ACLs (as opposed to fallback ACLs),
which means there are two cases:
- a Resource ACL already exists, and then it's just a matter of
updating it
- no Resource ACL exists, and the Resource is currently goverened by its
Fallback ACL. In this case, the Fallback ACL is copied, set as a new
Resource ACL, and then updated with the new access.

Note that the tests partly test for implementation by checking if the
underlying ACL functions are called, because many edge cases are already
ested for in that section of the code base,  and duplicating all these
tests would be very cumbersome.

# Checklist

- [X] All acceptance criteria are met.
- [X] Relevant documentation, if any, has been written/updated.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/access/wac.test.ts [hunk 1]
```diff
@@ -2375,3 +2377,714 @@ describe("setAgentAccess", () => {
...
+
+    expect(newAccess).toStrictEqual({
+      read: true,
+      append: true,
+      write: false,
+      control: false,
+    });
+  });
+
+  it("removes access previoulsy granted to the group if the new access is set to false", async () => {
```
Comment (Vinnl) [inline src/access/wac.test.ts:2709]:
```suggestion
  it("removes access previously granted to the group if the new access is set to false", async () => {
```

File: src/access/wac.ts [hunk 2]
```diff
@@ -408,3 +409,40 @@ export async function setAgentResourceAccess<T extends WithServerResourceInfo>(
...
+ * Access was regulated by its Fallback ACL, said Fallback ACL is copied to create
+ * a new Resource ACL. This has the side effect that the next time the Fallback
+ * ACL is updated, the changes _will not impact_ the target resource.
+ *
+ * If the target Resource's Access mode cannot be determined, e.g. the user does
+ * not have Read and Write access to the target Resource's ACL, or to its
+ * fallback ACL if it does not have a Resource ACL, then `null` is returned.
+ *
+ * @param resource The Resource for which Access is being set
+ * @param agent The Group for whom Access is being set
```
Comment (Vinnl) [inline src/access/wac.ts:426]:
```suggestion
 * @param agent The Group for which Access is being set
```

File: src/access/wac.ts [hunk 3]
```diff
@@ -408,3 +409,40 @@ export async function setAgentResourceAccess<T extends WithServerResourceInfo>(
...
+ * @param access The Access being set
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns The Resource, with its ACL updated, or null if the new Access could not
+ * be set.
+ */
+export async function setGroupResourceAccess<T extends WithServerResourceInfo>(
```
Comment (Vinnl) [inline src/access/wac.ts:432]: As in the other review I'd call this

```suggestion
export async function setGroupAccess<T extends WithServerResourceInfo>(
```

**Review by Vinnl — APPROVED** [review]

File: src/access/wac.ts (hunk 3 above)
Comment (NSeydoux) [reply src/access/wac.ts:387]: Same as for the previous PR #807, I think this should remain `setGroupResourceAccess`, and we can see how to design the default access in the universal API
